### PR TITLE
Adds preference on Android to select default Leo model

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/35560

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

The way that is described in the issue is the correct way how it should work. The in tab Leo page doesn't change default model.
What we missed is a setting to change a default model. That PR introduces it:
Screenshot_20240205_101209
Screenshot_20240205_101159
Go to Settings->Leo->Default model for new conversations.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPrefUtils.java [hunk 1]
```diff
@@ -79,4 +79,22 @@ public static void setChatProductId(String productId) {
         }
         UserPrefs.get(profileToUse).setString(BravePref.BRAVE_CHAT_PRODUCT_ID_ANDROID, productId);
     }
+
+    public static String getDefaultModel() {
+        Profile profileToUse = BraveLeoPrefUtils.getProfile();
+        if (profileToUse == null) {
+            Log.e(TAG, "BraveLeoPrefUtils.setChatProductId profile is null");
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPrefUtils.java:86]: nit: here method name must be `BraveLeoPrefUtils.getDefaultModel`

File: android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPrefUtils.java [hunk 2]
```diff
@@ -79,4 +79,22 @@ public static void setChatProductId(String productId) {
...
+            return "";
+        }
+        return UserPrefs.get(profileToUse).getString(BravePref.DEFAULT_MODEL_KEY);
+    }
+
+    public static void setDefaultModel(String modelKey) {
+        Profile profileToUse = BraveLeoPrefUtils.getProfile();
+        if (profileToUse == null) {
+            Log.e(TAG, "BraveLeoPrefUtils.setChatProductId profile is null");
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPrefUtils.java:95]: the same, `BraveLeoPrefUtils.setDefaultModel`

**Review by AlexeyBarabash — APPROVED** [review]
++

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by samartnik — APPROVED** [review]
lgtm

**Review by deeppandya — APPROVED** [review]
LGTM

**Review by tapanmodh — APPROVED** [review]
lgtm

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 14 AP11.231215.009 release-keys
Brave build: 1.64.51 
Chromium: 121.0.6167.139 (Official Build) canary (64-bit) 
```

---------------------------

**STEPS:**
1. Install and launch Brave `1.64.51` or newer
2. Three-dot menu > Settings > Leo > Verify `Default model for new conversations` is set to `Mixtral`
3. Tap `Default model for new conversations` > Verify `Default model` is set to `Mixtral`
4. Close the settings and go to any website, for example, https://chromeunboxed.com
5. Three-dot menu > Leo
6. Tap `Summarize this page` > Accept and begin > Confirm that summarize is done
7. Tap `Three-dot menu` in the top right corner on the `Leo` page > Verify that `Mixtral` is set by default
8. Choose `Claude Instant` or `Llama 2 13b`
9. On the website tap and hold any link > Open in new tab/Open in new tab in group (or you can simply tap + and open another website)
10. On the newly opened tab tap the `Three-dot` menu > Leo
11. Three-dot menu on Leo's page > Verify it switched back to the default model which is set to `Mixtral`
12. Close `Leo` > Three-dot menu > Settings > Leo > Default model for new conversations
13. Switch it from `Mixtral` to `Claude Instant` or `Llama 2 13b` > Close the settings
14. Open another link in the new tab > Three-dot menu > Leo > Tree-dot menu > Verify that the language model is the same as the one in the `Leo` settings

**ACTUAL RESULTS:**
- Verified that `Default model for new conversations` section is shown in `Leo` settings
- Verified that `Default model` is set to `Mixtral`
- Verified that `Mixtral` is shown as the default language model on the `Leo` page
- Verified that changes in the language model on a particular page are not changing the actual default language model
- Verified that changing the default language model in Leo settings is changing the same on the newly opened Leo page

---------------------------

|1|2|3|4|
|:-:|:-:|:-:|:-:|
|  |  |  |  |
|  |  |  |  |

https://github.com/brave/brave-core/assets/128532965/df5dbd98-afac-4288-be1c-0d722849026a

**Pull request merged**